checks/cs50/2017/x/speller/checks.py

```diff
@@ -0,0 +1,58 @@
+import os
+import sys
+
+sys.path.append(os.getcwd())
+from check50 import *
+
+class Speller(Checks):
+
+    @check()
+    def exists(self):
+        """dictionary.c, dictionary.h, and Makefile exist"""
+        super(Speller, self).exists("dictionary.c")
+        super(Speller, self).exists("dictionary.h")
+        super(Speller, self).exists("Makefile")
+
+    @check("exists")
+    def compiles(self):
+        """speller compiles"""
+        self.include("speller.c")
+        self.spawn("make").exit(0)
+
+    @check("compiles")
+    def basic(self):
+        """handles most basic words properly"""
+        self.include("basic")
+        self.spawn("./speller basic/dict basic/text").stdout(File("basic/out")).exit(0)
```

Comment:
Issue here is that we're subclassing `unittest.TestCase` which itself already has a method call `run` whose semantics is "run the test". That `run` method gets called when a `unittest.TestSuite` is running, so if we name our function `self.run`, we override the existing `run` method and `unittest` gets confused. So we needed something other than `run` as the name of the function, and `spawn` made the most sense to be consistent with `pexpect.spawn`.